packages/@aws-cdk/core/lib/cfn-parameter.ts

```diff
@@ -107,17 +117,166 @@ export class CfnParameter extends CfnElement {
+   */
+  public get allowedPattern(): string | undefined {
+    return this._allowedPattern;
+  }
+
+  public set allowedPattern(pattern: string | undefined) {
+    this._allowedPattern = pattern;
+  }
+
+  /**
+   * An array containing the list of values allowed for the parameter.
+   *
+   * @default - No constraints on values allowed for parameter.
+   */
+
+  public get allowedValues(): string[] | undefined {
+    return this._allowedValues;
+  }
+
+  public set allowedValues(values: string[] | undefined) {
+    this._allowedValues = values;
+  }
+
+  /**
+   * A string that explains a constraint when the constraint is violated.
+   * For example, without a constraint description, a parameter that has an allowed
+   * pattern of [A-Za-z0-9]+ displays the following error message when the user specifies
+   * an invalid value:
+   *
+   * @default - No description with customized error message when user specifies invalid values.
+   */
+  public get constraintDescription(): string | undefined {
+    return this._constraintDescription;
+  }
+
+  public set constraintDescription(desc: string | undefined) {
+    this._constraintDescription = desc;
+  }
+
+  /**
+   * A string of up to 4000 characters that describes the parameter.
+   *
+   * @default - No description for the parameter.
+   */
+  public get description(): string | undefined {
+    return this._description;
+  }
+
+  public set description(desc: string | undefined) {
+    this._description = desc;
+  }
+
+  /**
+   * An integer value that determines the largest number of characters you want to allow for String types.
+   *
+   * @default - None.
+   */
+  public get maxLength(): number | undefined {
+    return this._maxLength;
+  }
+
+  public set maxLength(len: number | undefined) {
+    this._maxLength = len;
+  }
+
+  /**
+   * An integer value that determines the smallest number of characters you want to allow for String types.
+   *
+   * @default - None.
+   */
+  public get minLength(): number | undefined {
+    return this._minLength;
+  }
+
+  public set minLength(len: number | undefined) {
+    this._minLength = len;
+  }
+
+  /**
+   * A numeric value that determines the largest numeric value you want to allow for Number types.
+   *
+   * @default - None.
+   */
+  public get maxValue(): number | undefined {
+    return this._maxValue;
+  }
+
+  public set maxValue(len: number | undefined) {
+    this._maxValue = len;
+  }
+  /**
+   * A numeric value that determines the smallest numeric value you want to allow for Number types.
+   *
+   * @default - None.
+   */
+  public get minValue(): number | undefined {
+    return this._minValue;
+  }
+
+  public set minValue(len: number | undefined) {
+    this._minValue = len;
   }
 
   /**
    * Indicates if this parameter is configured with "NoEcho" enabled.
    */
   public get noEcho(): boolean {
-    return !!this.props.noEcho;
+    return !!this._noEcho;
+  }
+
+  public set noEcho(echo: boolean) {
```

Comment:
I believe the setter should take `boolean | undefined` (if you just want to rely on the CFN default).